### PR TITLE
Compute duration for download and other events

### LINKED ISSUE #946: Add duration metric to download
Also look at other telemetry for places we need duration

### PR DESCRIPTION
**Pull request by boumenot**
Closes #946

### CODE REVIEW

File: bin/waagent2.0 [hunk 1]
```diff
@@ -54,6 +54,8 @@ import datetime
 import xml.sax.saxutils
 from distutils.version import LooseVersion
 
+from azurelinuxagent.common.event import elapsed_milliseconds
```
Comment (hglkrijger) [inline bin/waagent2.0]: You can revert all these changes, no need to update 2.0

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -647,12 +648,13 @@ def copy_status_files(self, old_ext_handler_i):
...
 
-    def report_event(self, message="", is_success=True):
+    def report_event(self, message="", is_success=True, duration=0):
         version = self.ext_handler.properties.version
         add_event(name=self.ext_handler.name, version=version, message=message, 
-                  op=self.operation, is_success=is_success)
+                  op=self.operation, is_success=is_success, duration=duration)
 
     def download(self):
+        begin_utc = datetime.datetime.utcnow()
```
Comment (hglkrijger) [inline azurelinuxagent/ga/exthandlers.py]: For the EG data, I am not sure if this is sufficient. We may need to have start and end events for frame analysis. Could you confirm?

Comment (boumenot) [reply azurelinuxagent/ga/exthandlers.py]: EG uses the timestamp and the duration.

A start and end event would be necessary if the event never completes or is interrupted.  If that did happen you would never get the event.  If you had a start and end event you would be able to see a start without a end.  I didn't want to double the number of events sent quite yet, which is why I started with this.

Comment (hglkrijger) [reply azurelinuxagent/ga/exthandlers.py]: Ok, sounds good

**Comment by boumenot** [comment]
I reverted all of the changes on bin/waagen2.0.

**Pull request merged**